<a href="https://colab.research.google.com/github/DIANAFG07/EjerciciosAlura/blob/python/Ejercicios.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# MANIPULANDO DATOS EN PYTHON

### Recopilación y muestra de datos

1 - Podemos solicitar el nombre con la función input y asignar el resultado de la salida a una variable. A continuación, podemos imprimir el resultado de la variable en la función print usando la formatación f-string.

In [ ]:
nombre = input("Por favor ingrese su nombre: ")

Por favor ingrese su nombre: Diana


In [ ]:
print(f"El nombre ingresado fue {nombre}.")

El nombre ingresado fue Diana.


2 - Crea un programa que solicite al usuario que escriba su nombre y edad, y luego imprima "Hola, [nombre], tienes [edad] años."

In [ ]:
nombre = input("Escribe tu nombre: ")
edad = int(input("Escribe tu edad: "))

Escribe tu nombre: Diana
Escribe tu edad: 29


In [ ]:
print(f"Hola {nombre}, tienes {edad} años.")

Hola Diana, tienes 29 años.


3 - Crea un programa que solicite al usuario que escriba su nombre, edad y altura en metros, y luego imprima "Hola, [nombre], tienes [edad] años y mides [altura] metros."

In [ ]:
nombre = input("Escribe tu nombre: ")
edad = int(input("Escribe tu edad: "))
altura = float(input("Escribe tu altura en metros: "))

Escribe tu nombre: Diana
Escribe tu edad: 29
Escribe tu altura en metros: 1.50


In [ ]:
print(f"Hola {nombre}, tienes {edad} años y mides {altura} metros.")

Hola Diana, tienes 29 años y mides 1.5 metros.


### Calculadora con operadores

4 - Crea un programa que solicite dos valores numéricos al usuario y luego imprima la suma de ambos valores.

In [ ]:
numero1 = int(input("Ingrese el primer número: "))
numero2 = int(input("Ingrese el segundo número: "))

Ingrese el primer número: 10
Ingrese el segundo número: 2


In [ ]:
suma = numero1 + numero2
print(f"La suma del número {numero1} más {numero2} es {suma}")

La suma del número 10 más 2 es 12


5 - Crea un programa que solicite tres valores numéricos al usuario y luego imprima la suma de los tres valores.

In [ ]:
numero1 = int(input("Ingrese el primer número: "))
numero2 = int(input("Ingrese el segundo número: "))
numero3 = int(input("Ingrese el tercer número: "))

Ingrese el primer número: 2
Ingrese el segundo número: 2
Ingrese el tercer número: 2


In [ ]:
suma = numero1 + numero2 + numero3
print(f"La suma del número {numero1} más {numero2} más {numero3} es {suma}")

La suma del número 2 más 2 más 2 es 6


6 - Crea un programa que solicite dos valores numéricos al usuario y luego imprima la resta del primero menos el segundo valor.

In [ ]:
numero1 = int(input("Ingrese el primer número: "))
numero2 = int(input("Ingrese el segundo número: "))

Ingrese el primer número: 10
Ingrese el segundo número: 2


In [ ]:
resta = numero1 - numero2
print(f"La resta del número {numero1} menos {numero2} es {resta}")

La resta del número 10 menos 2 es 8


7 - Crea un programa que solicite dos valores numéricos al usuario y luego imprima la multiplicación de los dos valores.

In [ ]:
numero1 = int(input("Ingrese el primer número: "))
numero2 = int(input("Ingrese el segundo número: "))

Ingrese el primer número: 4
Ingrese el segundo número: 4


In [ ]:
multiplicacion = numero1 * numero2
print(f"La multiplicación del número {numero1} por {numero2} es {multiplicacion}")

La multiplicación del número 4 por 4 es 16


8 - Crea un programa que solicite dos valores numéricos, un numerador y un denominador, y realice la división entre los dos valores. Asegúrate de que el valor del denominador no sea igual a 0.

In [ ]:
numerador = int(input("Ingrese el número numerador: "))
denominador = int(input("Ingrese el número denominador, el cual debe de ser diferente de cero: "))

Ingrese el número numerador: 10
Ingrese el número denominador, el cual debe de ser diferente de cero: 3


In [ ]:
division = numerador / denominador
print(f"La división del número {numerador} y {denominador} es {division}")

La división del número 10 y 3 es 3.3333333333333335


9 - Crea un programa que solicite dos valores numéricos, un operador y una potencia, y realice la exponenciación entre estos dos valores.

In [ ]:
operador = int(input("Ingrese el número: "))
potencia = int(input("Ingrese la potencia: "))

Ingrese el número: 20
Ingrese la potencia: 3


In [ ]:
exponencia = operador ** potencia
print(f"La exponencia del número {operador} a la {potencia} potencia es {exponencia}")

La exponencia del número 20 a la 3 potencia es 8000


10 - Crea un programa que solicite dos valores numéricos, un numerador y un denominador, y realice la división entera entre los dos valores. Asegúrate de que el valor del denominador no sea igual a 0.

In [ ]:
numerador = int(input("Ingrese el número numerador: "))
denominador = int(input("Ingrese el número denominador, el cual debe de ser diferente de cero: "))

Ingrese el número numerador: 10
Ingrese el número denominador, el cual debe de ser diferente de cero: 3


In [ ]:
division = numerador // denominador
print(f"La división entera del número {numerador} y {denominador} es {division}")

La división entera del número 10 y 3 es 3


11 - Crea un programa que solicite dos valores numéricos, un numerador y un denominador, y devuelva el resto de la división entre los dos valores. Asegúrate de que el valor del denominador no sea igual a 0.

In [ ]:
numerador = int(input("Ingrese el número numerador: "))
denominador = int(input("Ingrese el número denominador, el cual debe de ser diferente de cero: "))

Ingrese el número numerador: 10
Ingrese el número denominador, el cual debe de ser diferente de cero: 3


In [ ]:
division = numerador % denominador
print(f"La resto de la división del número {numerador} y {denominador} es {division}")

La resto de la división del número 10 y 3 es 1


12 - Crea un código que solicite las 3 notas de un estudiante e imprima el promedio de las notas.

In [ ]:
nota1 = int(input("Ingrese la primer nota: "))
nota2 = int(input("Ingrese la segunda nota: "))
nota3 = int(input("Ingrese la tercer nota: "))

Ingrese la primer nota: 10
Ingrese la segunda nota: 9
Ingrese la tercer nota: 8


In [ ]:
promedio = (nota1 + nota2 + nota3) / 3
print(f"El promedio es {promedio}.")

El promedio es 9.0.


13 - Crea un código que calcule e imprima el promedio ponderado de los números 5, 12, 20 y 15 con pesos respectivamente iguales a 1, 2, 3 y 4.

In [ ]:
media_ponderada = (5*1 + 12*2 + 20*3 + 15*4) / (1+2+3+4)
print(f'Media {media_ponderada}.')

Media 14.9.


### Editando textos



14 - Crea una variable llamada "frase" y asígnale una cadena de texto de tu elección. Luego, imprime la frase en pantalla.

In [ ]:
frase = "Aquí debería de poner una buena frase."
print(frase)

Aquí debería de poner una buena frase.


15 - Crea un código que solicite una frase y luego imprima la frase en pantalla.

In [ ]:
frase = input("Escriba su frase: ")
print(frase)

Escriba su frase: Aquí debería de poner una buena frase.
Aquí debería de poner una buena frase.


16 - Crea un código que solicite una frase al usuario y luego imprima la misma frase ingresada pero en mayúsculas.

In [ ]:
frase = input("Escriba su frase: ")
print(frase.upper())

Escriba su frase: aqui deberia de poner una frase
AQUI DEBERIA DE PONER UNA FRASE


17 - Crea un código que solicite una frase al usuario y luego imprima la misma frase ingresada pero en minúsculas.

In [ ]:
frase = input("Escriba su frase: ")
print(frase.lower())

Escriba su frase: AQUI DEBERIA DE PONER UNA FRASE
aqui deberia de poner una frase


18 - Crea una variable llamada "frase" y asígnale una cadena de texto de tu elección. Luego, imprime la frase sin espacios en blanco al principio y al final.

In [ ]:
frase = "       Aquí debería de poner una buena frase.              "
print(frase.strip())

Aquí debería de poner una buena frase.


19 - Crea un código que solicite una frase al usuario y luego imprima la misma frase sin espacios en blanco al principio y al final.

In [ ]:
frase = input("Escriba su frase: ")
print(frase.strip())

Escriba su frase:        Aquí debería de poner una buena frase.              
Aquí debería de poner una buena frase.


20 - Crea un código que solicite una frase al usuario y luego imprima la misma frase sin espacios en blanco al principio y al final, además de convertirla a minúsculas.

In [ ]:
frase = input("Escriba su frase: ")
print(frase.lower().strip())

Escriba su frase:                  AQUI DEBERIA DE PONER UNA FRASE                 
aqui deberia de poner una frase


21 - Crea un código que solicite una frase al usuario y luego imprima la misma frase con todas las vocales "e" reemplazadas por la letra "f".

In [ ]:
frase = input("Escriba su frase: ")
print(frase.replace("e","f"))

Escriba su frase: aqui deberia de poner una frase
aqui dfbfria df ponfr una frasf


22 - Crea un código que solicite una frase al usuario y luego imprima la misma frase con todas las vocales "a" reemplazadas por el carácter "@".

In [ ]:
frase = input("Escriba su frase: ")
print(frase.replace("a","@"))

Escriba su frase: aqui deberia de poner una frase
@qui deberi@ de poner un@ fr@se


23 - Crea un código que solicite una frase al usuario y luego imprima la misma frase con todas las consonantes "s" reemplazadas por el carácter "$".

In [ ]:
frase = input("Escriba su frase: ")
print(frase.replace("s","$"))

Escriba su frase: Las fresas son muy ricas
La$ fre$a$ $on muy rica$


# ESTRUCTURAS CONDICIONALES

### If Elif Else

1 - Escribe un programa que pida a la persona usuaria que proporcione dos números y muestre el número más grande.

In [ ]:
# Recolectamos los números
num1 = float(input('Ingrese el primer número: '))
num2 = float(input('Ingrese el segundo número: '))
# Comparamos ambos números y determinamos cuál es el mayor
if num1 > num2:
    print(f'El primer número es mayor: {num1}')
elif num2 > num1:
    print(f'El segundo número es mayor: {num2}')
else: # En caso de que los números sean iguales
    print('Ambos números son iguales.')

Ingrese el primer número: 2.5
Ingrese el segundo número: 6.8
El segundo número es mayor: 6.8


2 - Escribe un programa que solicite el porcentaje de crecimiento de producción de una empresa e informe si hubo un crecimiento (porcentaje positivo) o una disminución (porcentaje negativo).

In [ ]:
# Recolectamos el porcentaje
variación = float(input('Ingrese el porcentaje de crecimiento: '))
# Verificamos si el valor es positivo o negativo con una comparación para ver si el número
# es mayor o menor que 0
if variación > 0:
    print(f'Hubo un crecimiento del {variación}%')
elif variación < 0:
    print(f'Hubo un decrecimiento del {variación}%')
else:
    print('No hubo crecimiento ni decrecimiento.')

Ingrese el porcentaje de crecimiento: 4
Hubo un crecimiento del 4.0%


3 - Podemos verificar si una letra es una vocal o una consonante al comprobar si el carácter está contenido en una cadena de vocales utilizando el operador in.

In [ ]:
# Recolectamos la letra del usuario en minúsculas
letra = input('Ingrese una letra: ').lower()
vocales = 'aeiou' # cadena que contiene todas las vocales
# Verificamos si la letra está en las vocales con `in`
if letra in vocales:
    print('La letra es una vocal.')
else:
    print('La letra es una consonante.')

Ingrese una letra: d
La letra es una consonante.


4 - Comparamos cada valor con los otros dos valores correspondientes a los otros dos años y determinamos el valor más alto y el más bajo. Lo hacemos asignando inicialmente el valor de precio_ano1 como el valor máximo y, si encontramos un valor mayor, actualizamos la variable mayor. Utilizamos una lógica similar para encontrar el valor mínimo.

In [ ]:
# Recolectamos los precios de los 3 años
precio_ano1 = float(input('Ingrese el precio promedio del automóvil en el primer año: '))
precio_ano2 = float(input('Ingrese el precio promedio del automóvil en el segundo año: '))
precio_ano3 = float(input('Ingrese el precio promedio del automóvil en el tercer año: '))
# Determinamos el valor más alto mediante comparaciones
mayor = precio_ano1
if precio_ano2 > mayor:
  mayor = precio_ano2
if precio_ano3 > mayor:
  mayor = precio_ano3
# Determinamos el valor más bajo mediante comparaciones
menor = precio_ano1
if precio_ano2 < menor:
  menor = precio_ano2
if precio_ano3 < menor:
  menor = precio_ano3
# Mostramos los resultados
print(f'El precio más alto fue de R$ {mayor}.')
print(f'El precio más bajo fue de R$ {menor}.')

Ingrese el precio promedio del automóvil en el primer año: 300000
Ingrese el precio promedio del automóvil en el segundo año: 257000.43
Ingrese el precio promedio del automóvil en el tercer año: 233000.97
El precio más alto fue de R$ 300000.0.
El precio más bajo fue de R$ 233000.97.


5.-

In [ ]:
# Recolectamos el precio de tres productos
producto1 = float(input('Ingrese el precio del primer producto: '))
producto2 = float(input('Ingrese el precio del segundo producto: '))
producto3 = float(input('Ingrese el precio del tercer producto: '))

# Usamos el operador lógico `and` para determinar cuál es el precio más bajo entre los 3 productos
# ya que esto nos permite hacer una comparación de 3 entradas
if producto1 < producto2 and producto1 < producto3:
    print('El primer producto es el más barato.')
elif producto2 < producto1 and producto2 < producto3:
    print('El segundo producto es el más barato.')
else:
    print('El tercer producto es el más barato.')

Ingrese el precio del primer producto: 23
Ingrese el precio del segundo producto: 21
Ingrese el precio del tercer producto: 43
El segundo producto es el más barato.


6 - Después de recolectar los 3 números, realizamos comparaciones siguiendo una lógica similar a la pregunta anterior. Utilizamos el operador lógico and para determinar cuál es el número más grande entre los 3 datos de productos, luego verificamos entre los dos más pequeños y finalmente utilizamos print para mostrar los números en orden descendente mediante varias declaraciones condicionales anidadas.

In [ ]:
# Recolectamos los 3 números
num1 = int(input('Ingrese el primer número: '))
num2 = int(input('Ingrese el segundo número: '))
num3 = int(input('Ingrese el tercer número: '))

# Comparación entre los 3 números
if (num1 >= num2) and (num1 >= num3):
    print(num1)
    if num2 >= num3:
        print(num2)
        print(num3)
    else:
        print(num3)
        print(num2)
elif (num2 >= num1) and (num2 >= num3):
    print(num2)
    if num1 >= num3:
        print(num1)
        print(num3)
    else:
        print(num3)
        print(num1)
else:
    print(num3)
    if num1 >= num2:
        print(num1)
        print(num2)
    else:
        print(num2)
        print(num1)

Ingrese el primer número: 2
Ingrese el segundo número: 3
Ingrese el tercer número: 4
4
3
2


7 -Escribe un programa que pregunte en qué turno estudia la persona usuaria ("mañana", "tarde" o "noche") y muestre el mensaje "¡Buenos Días!", "¡Buenas Tardes!", "¡Buenas Noches!" o "Valor Inválido!", según el caso.

In [ ]:
# Recolectamos el turno de estudio
turno = input('Ingresa en qué turno estudias (mañana, tarde o noche): ')

# Comparamos la entrada con todas las opciones y mostramos el resultado.
if turno == 'mañana':
  print('¡Buenos Días!')
elif turno == 'tarde':
  print('¡Buenas Tardes!')
elif turno == 'noche':
  print('¡Buenas Noches!')
else:
  print('¡Valor Inválido!')

Ingresa en qué turno estudias (mañana, tarde o noche): tarde
¡Buenas Tardes!


8 - Podemos usar el operador de módulo % para determinar si un número es par o impar. Si la división entera de un número entre 2 da como resultado 0, entonces es par. Si no, es impar. Esto se debe a que todos los números pares son divisibles por 2, por lo que no tienen un residuo en la división.

In [ ]:
# Recolectamos los datos
num = int(input('Ingresa un número: '))

# Verificamos si el número es par según el resultado del módulo
if num % 2 == 0:
    print('El número es par.')
else:
    print('El número es impar.')

Ingresa un número: 3
El número es impar.


9 - Podemos usar el operador de módulo % para determinar si un número es entero o decimal. Si el operador de módulo % devuelve cero en la división entera de un número entre 1, entonces es un número entero. De lo contrario, es un número decimal.

In [ ]:
# Recolectamos los datos
num = float(input('Ingresa un número: '))
# Verificamos si el número es entero o decimal según el resultado del módulo
if num % 1 == 0:
    print('El número es entero.')
else:
    print('El número es decimal.')

Ingresa un número: 2
El número es entero.


10 - Un programa debe ser escrito para leer dos números y luego preguntar a la persona usuaria qué operación desea realizar. El resultado de la operación debe incluir información sobre el número, si es par o impar, positivo o negativo, e entero o decimal.

In [ ]:
# Recolectamos los números a operar y solicitamos la operación deseada por el usuario
num1 = float(input('Ingrese el primer número: '))
num2 = float(input('Ingrese el segundo número: '))
operación = input('Ingrese la operación deseada (+, -, *, /): ')

# Verificamos la operación seleccionada y realizamos la operación matemática según la elección
if operación == '+':
    resultado = num1 + num2
elif operación == '-':
    resultado = num1 - num2
elif operación == '*':
    resultado = num1 * num2
elif operación == '/':
    resultado = num1 / num2
else: # Especificamos un resultado en caso de que el usuario no ingrese una de las operaciones correctamente.
    print('Operación no válida, el resultado de la operación será 0')
    resultado = 0

# Realizamos las mismas verificaciones que en preguntas anteriores para generar el informe del cálculo entre números
if resultado % 1 == 0:
    print('El resultado es un número entero.')
else:
    print('El resultado es un número decimal.')

if resultado > 0:
    print('El resultado es positivo.')
else:
    print('El resultado es negativo.')

if resultado % 2 == 0:
    print('El resultado es un número par.')
else:
    print('El resultado es un número impar.')

Ingrese el primer número: 2
Ingrese el segundo número: 10
Ingrese la operación deseada (+, -, *, /): /
El resultado es un número decimal.
El resultado es positivo.
El resultado es un número impar.


11 - Después de recolectar los valores de los 3 lados de un triángulo, debemos verificar si realmente pueden formar un triángulo siguiendo el consejo "Tres lados forman un triángulo cuando la suma de cualquier par de lados es mayor que el tercer lado". Esta verificación se puede realizar con el operador "and". Luego, podemos verificar si todos los lados son iguales, lo que forma un triángulo equilátero, o si todos los lados son diferentes, lo que forma un triángulo escaleno. Estas verificaciones se pueden realizar con el operador "and" y los operadores "== y "! =". Finalmente, utilizamos "else" para el caso de un triángulo isósceles.

In [ ]:
# Recolectamos los lados de un triángulo
print('Estamos recopilando los lados de un triángulo.')
lado1 = float(input('Ingresa la longitud del primer lado: '))
lado2 = float(input('Ingresa la longitud del segundo lado: '))
lado3 = float(input('Ingresa la longitud del tercer lado: '))

# Verificamos si los lados pueden formar un triángulo
if (lado1 + lado2 > lado3) and (lado2 + lado3 > lado1) and (lado1 + lado3 > lado2):
    print('¡Los valores pueden formar un triángulo!')
    # Comparamos los lados para determinar el tipo de triángulo
    if (lado1 == lado2) and (lado2 == lado3):
        print('El triángulo es equilátero.')
    elif (lado1 != lado2) and (lado2 != lado3) and (lado1 != lado3):
        print('El triángulo es escaleno.')
    else:
        print('El triángulo es isósceles.')
else:
    print('¡Los valores no pueden formar un triángulo!')

Estamos recopilando los lados de un triángulo.
Ingresa la longitud del primer lado: 10
Ingresa la longitud del segundo lado: 2
Ingresa la longitud del tercer lado: 5
¡Los valores no pueden formar un triángulo!


12 - Un establecimiento está vendiendo combustibles con descuentos variables. Para el etanol, si la cantidad comprada es de hasta 15 litros, el descuento será del 2% por litro. En caso contrario, será del 4% por litro. Para el diésel, si la cantidad comprada es de hasta 15 litros, el descuento será del 3% por litro. En caso contrario, será del 5% por litro. El precio por litro de diésel es de R$ 2,00 y el precio por litro de etanol es de R$ 1,70. Escribe un programa que lea la cantidad de litros vendidos y el tipo de combustible (E para etanol y D para diésel) y calcule el valor a pagar por el cliente.

In [ ]:
# Recolectamos la cantidad de litros y el tipo de combustible,
# convirtiendo el carácter en mayúsculas para facilitar nuestro análisis
cantidad_litros = float(input('Ingrese la cantidad de litros vendidos: '))
tipo_combustible = input('Ingrese el tipo de combustible (E para etanol y D para diésel): ').upper()

# Verificamos primero el tipo de combustible
if tipo_combustible == 'E':
  # Establecemos el precio por litro de etanol
  precio_litro = 1.70
  # Según la cantidad de litros, establecemos el descuento correspondiente
  if cantidad_litros <= 15:
    descuento = 0.02
  else:
    descuento = 0.04
elif tipo_combustible == 'D':
  # Establecemos el precio por litro de diésel
  precio_litro = 2.00
  # Según la cantidad de litros, establecemos el descuento correspondiente
  if cantidad_litros <= 15:
    descuento = 0.03
  else:
    descuento = 0.05
# En caso de error en la especificación del tipo de combustible,
# consideramos las entradas como no válidas y establecemos los precios y descuentos en 0
else:
    print('Entradas no válidas!')
    precio_litro = 0
    descuento = 0

# Calculamos el valor del descuento, seguido del cálculo del precio descontado
valor_descuento = precio_litro * cantidad_litros * descuento
valor_pagado = precio_litro * cantidad_litros - valor_descuento

# Resultado
print(f'Valor a pagar por el cliente: R$ {valor_pagado}')

Ingrese la cantidad de litros vendidos: 10
Ingrese el tipo de combustible (E para etanol y D para diésel): D
Valor a pagar por el cliente: R$ 19.4


13 - En una empresa de venta de bienes raíces, debes crear un código que analice los datos de ventas anuales para ayudar a la dirección en la toma de decisiones. El código debe recopilar los datos de cantidad de ventas durante los años 2022 y 2023 y calcular la variación porcentual.

In [ ]:
# Recolectamos las ventas de los dos años
venta_2022 = float(input('Ingrese la cantidad de ventas en 2022: '))
venta_2023 = float(input('Ingrese la cantidad de ventas en 2023: '))

# Calculamos la variación porcentual entre las ventas de los años 2022 y 2023
var_porcentual = 100 * (venta_2023 - venta_2022) / (venta_2022)

# Análisis condicional de la variación porcentual para determinar la sugerencia a enviar
if var_porcentual > 20:
    print('Bonificación para el equipo de ventas.')
elif 2 <= var_porcentual <= 20:
    print('Pequeña bonificación para el equipo de ventas.')
elif -10 <= var_porcentual < 2:
    print('Planificación de políticas de incentivo a las ventas.')
else:
    print('Recorte de gastos.')

Ingrese la cantidad de ventas en 2022: 200000
Ingrese la cantidad de ventas en 2023: 180000
Planificación de políticas de incentivo a las ventas.


# ESTRUCTURAS DE REPETICIÓN

1 - Escribe un programa que solicite dos números enteros e imprima todos los números enteros entre ellos.

2 - Escribe un programa para calcular cuántos días tomará que la colonia de una bacteria A supere o iguale a la colonia de una bacteria B, basado en tasas de crecimiento del 3% y 1.5%, respectivamente. Supón que la colonia A comienza con 4 elementos y B con 10.

3 - Para procesar una cantidad de 15 datos de evaluaciones de usuarios de un servicio de la empresa, necesitamos verificar si las calificaciones son válidas. Por lo tanto, escribe un programa que recibirá calificaciones del 0 al 5 y verificará si son valores válidos. Si se ingresa una calificación superior a 5 o inferior a 0, se repetirá hasta que el usuario ingrese un valor válido.

4 - Desarrolla un programa que lea un conjunto indefinido de temperaturas en grados Celsius y calcule su promedio. La lectura debe detenerse al ingresar el valor -273°C.

5 - Escribe un programa que calcule el factorial de un número entero proporcionado por el usuario. Recuerda que el factorial de un número entero es el producto de ese número por todos sus antecesores hasta llegar al número 1. Por ejemplo, el factorial de 5 es 5 x 4 x 3 x 2 x 1 = 120.

Momento de los proyectos

6 - Escribe un programa que genere la tabla de multiplicar de un número entero del 1 al 10, según la elección del usuario. Como ejemplo, para el número 2, la tabla de multiplicar debe mostrarse en el siguiente formato:

Tabla de multiplicar del 2:
2 x 1 = 2
2 x 2 = 4
[...]
2 x 10 = 20
Copia el código
7 - Los números primos tienen diversas aplicaciones en Ciencia de Datos, como en criptografía y seguridad. Un número primo es aquel que es divisible solo por sí mismo y por 1. Por lo tanto, crea un programa que solicite un número entero y determine si es un número primo o no.

8 - Vamos a comprender la distribución de edades de los pensionistas de una empresa de seguros. Escribe un programa que lea las edades de una cantidad no informada de clientes y muestre la distribución en los intervalos [0-25], [26-50], [51-75] y [76-100]. La entrada de datos se detendrá al ingresar un número negativo.

9 - En una elección para la gerencia de una empresa con 20 empleados, hay cuatro candidatos. Escribe un programa que calcule al ganador de la elección. La votación se realizó de la siguiente manera:

Cada empleado votó por uno de los cuatro candidatos (representados por los números 1, 2, 3 y 4).

También se contaron los votos nulos (representados por el número 5) y los votos en blanco (representados por el número 6).

Al final de la votación, el programa debe mostrar el número total de votos para cada candidato, los votos nulos y los votos en blanco. Además, debe calcular y mostrar el porcentaje de votos nulos con respecto al total de votos y el porcentaje de votos en blanco con respecto al total de votos.

Si necesitas ayuda, las soluciones a las actividades están disponibles en la sección "Opinión del instructor".

Ver opinión del instructor
Opinión del instructor

Pueden existir diversas formas de resolver los problemas propuestos, y algunas de ellas se presentan a continuación.

Entrenando la programación

1 -

# Recolectamos los valores de inicio y fin
inicio = int(input('Ingresa el primer número entero: '))
fin = int(input('Ingresa el segundo número entero: '))

# Verificamos si el valor de inicio es menor que el fin
if inicio < fin:
  # Podemos imprimir los enteros entre el valor menor y el valor mayor
  for i in range(inicio + 1, fin):
    print(i)
elif inicio > fin:
  for i in range(fin + 1, inicio):
    print(i)
else: # En caso de que los números sean iguales, no podemos imprimir ninguna secuencia.
  print('Los números son iguales.')
Copia el código
2 -

# Número inicial de bacterias
colonia_a = 4
colonia_b = 10

# Tasas de crecimiento de las colonias
tasa_a = 0.03
tasa_b = 0.015

# Contador de días
días = 0

# La condición que termina el bucle es cuando
# la colonia A supera a la colonia B
while colonia_a <= colonia_b:
  # Usamos un operador de asignación con multiplicación
  colonia_a *= 1 + tasa_a
  colonia_b *= 1 + tasa_b
  # Contamos los días en cada iteración
  días += 1

# Resultado final
print(f'Necesitará {días} días para que la colonia A supere a la colonia B.')
Copia el código
3 -

# Bucle para recopilar las 15 notas
for i in range(15):
  nota = float(input(f'Ingresa la nota del usuario {i}: ')

  # Verifica si la nota está entre 0 y 5
  # Si no lo está, el bucle se repetirá hasta que se obtenga un valor válido
  while (nota < 0) or (nota > 5):
    nota = float(input(f'Nota no válida, ingresa nuevamente la nota del usuario {i}: '))

print('Verificación completa. Todas las notas son válidas.')
Copia el código
4 -

# Recopilamos la temperatura
temperatura = float(input('Ingresa la temperatura en grados Celsius: '))

# Inicializamos un contador y una suma para calcular el promedio
contador = 0
suma = 0

# Nuestro código se ejecuta hasta que el valor de temperatura sea igual a -273
while temperatura != -273:
    # La suma se actualiza sumando la temperatura a la variable suma
    suma += temperatura
    # Contamos la cantidad de valores recopilados con el contador
    contador += 1
    # Recopilamos nuevamente la temperatura
    temperatura = float(input('Ingresa la temperatura en grados Celsius: '))

promedio = suma / contador

print(f'El promedio de las temperaturas es: {promedio}')
Copia el código
5 -

# Solicita el número
num = int(input('Ingresa un número entero: ')

# Inicializa el cálculo
factorial = 1

# Nuestro contador comienza con el número máximo
# y se realizará un conteo decreciente con el operador -=
i = num
while i > 0:
    # Queremos multiplicar el valor factorial por el número
    # y todos los números por debajo de él hasta 1
    factorial *= i
    i -= 1

# Imprime el cálculo del factorial
print(f'El factorial de {num} es {factorial}')

Momento de los proyectos
Copia el código
6 -

# Solicita el número
num = int(input('Ingresa un número entero del 1 al 10: ')

# Generamos la tabla de multiplicar
print(f'Tabla de multiplicar del {num}:')
for i in range(1, 11):
    resultado = num * i
    print(f'{num} x {i} = {resultado}')
Copia el código
7 -

# Recopilamos el número
num = int(input('Ingresa un número entero: ')

# Los números enteros iguales o menores que 1 no se consideran primos
if num > 1:
    for i in range(2, num):
        # Verificamos todos los residuos de la división entre todos los números menores que num
        # Si algún residuo es 0, significa que es divisible por otro número además de sí mismo y 1
        if (num % i) == 0:
            print(f'{num} no es un número primo')
            break
    else:
        print(f'{num} es un número primo')
else:
    print(f'{num} no es un número primo')
Copia el código
8 -

# Recopilamos las edades de los clientes
edad = int(input('Ingresa la edad (o un número negativo para finalizar): ')

# Inicializamos las variables de conteo
contador_0_25 = 0 # contador de edades entre 0 y 25
contador_26_50 = 0 # contador de edades entre 26 y 50
contador_51_75 = 0 # contador de edades entre 51 y 75
contador_76_100 = 0 # contador de edades entre 76 y 100

# Nuestro código se ejecuta hasta que el valor de edad sea negativo
while edad >= 0:
    # Contamos cada caso
    if edad >= 0 and edad <= 25:
        contador_0_25 += 1
    elif edad >= 26 and edad <= 50:
        contador_26_50 += 1
    elif edad >= 51 and edad <= 75:
        contador_51_75 += 1
    elif edad >= 76 and edad <= 100:
        contador_76_100 += 1
    
    # Repetimos el proceso de entrada de datos hasta que se ingrese un número negativo    
    edad = int(input('Ingresa la edad (o un número negativo para finalizar): ')

# Mostramos los resultados
print('Distribución de edades:')
print('[0-25]:', contador_0_25)
print('[26-50]:', contador_26_50)
print('[51-75]:', contador_51_75)
print('[76-100]:', contador_76_100)
Copia el código
9 -

# Inicializamos las variables contadoras
votos_candidato1 = 0
votos_candidato2 = 0
votos_candidato3 = 0
votos_candidato4 = 0
votos_nulos = 0
votos_blanco = 0

# Inicio del bucle para leer los votos
for i in range(0, 20):
    voto = int(input('Ingresa tu voto: ')
    if voto == 1:
        votos_candidato1 += 1
    elif voto == 2:
        votos_candidato2 += 1
    elif voto == 3:
        votos_candidato3 += 1
    elif voto == 4:
        votos_candidato4 += 1
    elif voto == 5:
        votos_nulos += 1
    elif voto == 6:
        votos_blanco += 1
    else:
        print("Voto inválido.")

print(f'Votos candidato 1: {votos_candidato1}')
print(f'Votos candidato 2: {votos_candidato2}')
print(f'Votos candidato 3: {votos_candidato3}')
print(f'Votos candidato 4: {votos_candidato4}')
print(f'Votos nulos: {votos_nulos}')
print(f'Votos en blanco: {votos_blanco}')
print(f'Porcentaje de votos nulos: {(votos_nulos / 20 * 100)}')
print(f'Porcentaje de votos en blanco: {(votos_blanco / 20 * 100)}')

1 - Crea un programa que tenga la siguiente lista con los gastos de una empresa de papel [2172.54, 3701.35, 3518.09, 3456.61, 3249.38, 2840.82, 3891.45, 3075.26, 2317.64, 3219.08]. Con estos valores, crea un programa que calcule el promedio de gastos. Sugerencia: usa las funciones integradas sum() y len().

2 - Con los mismos datos de la pregunta anterior, determina cuántas compras se realizaron por encima de 3000 reales y calcula el porcentaje con respecto al total de compras.

3 - Crea un código que recoja en una lista 5 números enteros aleatorios e imprima la lista. Ejemplo: [1, 4, 7, 2, 4].

4 - Recoge nuevamente 5 números enteros e imprime la lista en orden inverso al enviado.

5 - Crea un programa que, al ingresar un número cualquiera, genere una lista que contenga todos los números primos entre 1 y el número ingresado.

6 - Escribe un programa que pida una fecha, especificando el día, mes y año, y determine si es válida para su análisis.

Momento para los proyectos

7 - Para un estudio sobre la multiplicación de bacterias en una colonia, se recopiló el número de bacterias multiplicadas por día y se puede observar a continuación: [1.2, 2.1, 3.3, 5.0, 7.8, 11.3, 16.6, 25.1, 37.8, 56.9]. Con estos valores, crea un código que genere una lista que contenga el porcentaje de crecimiento de bacterias por día, comparando el número de bacterias en cada día con el número de bacterias del día anterior. Sugerencia: para calcular el porcentaje de crecimiento, utiliza la siguiente ecuación: 100 * (muestra_actual - muestra_anterior) / muestra_anterior.

8 - Para una selección de productos alimenticios, debemos separar el conjunto de IDs proporcionados por números enteros, sabiendo que los productos con ID par son dulces y los que tienen ID impar son amargos. Crea un código que recoja 10 IDs. Luego, calcula y muestra la cantidad de productos dulces y amargos.

9 - Desarrolla un programa que informe la puntuación de un estudiante de acuerdo con sus respuestas. Debe pedir la respuesta del estudiante para cada pregunta y verificar si la respuesta coincide con el resultado. Cada pregunta vale un punto y hay opciones A, B, C o D.

Resultado del examen:
01 - D
02 - A
03 - C
04 - B
05 - A
06 - D
07 - C
08 - C
09 - A
10 - B
Copia el código
10 - Un instituto de meteorología desea realizar un estudio de la temperatura media de cada mes del año. Para ello, debes crear un código que recoja y almacene esas temperaturas medias en una lista. Luego, calcula el promedio anual de las temperaturas y muestra todas las temperaturas por encima del promedio anual y en qué mes ocurrieron, mostrando los meses por su nombre (Enero, Febrero, etc.).

11 - Una empresa de comercio electrónico está interesada en analizar las ventas de sus productos. Los datos de ventas se han almacenado en un diccionario:

{'Producto A': 300, 'Producto B': 80, 'Producto C': 60, 'Producto D': 200, 'Producto E': 250, 'Producto F': 30}
Copia el código
Escribe un código que calcule el total de ventas y el producto más vendido.

12 - Se realizó una encuesta de mercado para decidir cuál diseño de marca infantil es más atractivo para los niños. Los votos de la encuesta se pueden ver a continuación:

Tabla de votos de la marca
Diseño 1 - 1334 votos
Diseño 2 - 982 votos
Diseño 3 - 1751 votos
Diseño 4 - 210 votos
Diseño 5 - 1811 votos
Copia el código
Adapta los datos proporcionados a una estructura de diccionario. A partir de ello, informa el diseño ganador y el porcentaje de votos recibidos.

13 - Los empleados de un departamento de tu empresa recibirán una bonificación del 10% de su salario debido a un excelente rendimiento del equipo. El departamento de finanzas ha solicitado tu ayuda para verificar las consecuencias financieras de esta bonificación en los recursos. Se te ha enviado una lista con los salarios que recibirán la bonificación: [1172, 1644, 2617, 5130, 5532, 6341, 6650, 7238, 7685, 7782, 7903]. La bonificación de cada empleado no puede ser inferior a 200. En el código, convierte cada uno de los salarios en claves de un diccionario y la bonificación de cada salario en el valor correspondiente. Luego, informa el gasto total en bonificaciones, cuántos empleados recibieron la bonificación mínima y cuál fue el valor más alto de la bonificación proporcionada.

14 - Un equipo de científicos de datos está estudiando la diversidad biológica en un bosque. El equipo recopiló información sobre el número de especies de plantas y animales en cada área del bosque y almacenó estos datos en un diccionario. En él, la clave describe el área de los datos y los valores en las listas corresponden a las especies de plantas y animales en esas áreas, respectivamente.

{'Área Norte': [2819, 7236], 'Área Leste': [1440, 9492], 'Área Sul': [5969, 7496], 'Área Oeste': [14446, 49688], 'Área Centro': [22558, 45148]}
Copia el código
Escribe un código para calcular el promedio de especies por área e identificar el área con la mayor diversidad biológica. Sugerencia: utiliza las funciones incorporadas sum() y len().

15 - El departamento de Recursos Humanos de tu empresa te pidió ayuda para analizar las edades de los colaboradores de 4 sectores de la empresa. Para ello, te proporcionaron los siguientes datos:

{'Setor A': [22, 26, 30, 30, 35, 38, 40, 56, 57, 65],
 'Setor B': [22, 24, 26, 33, 41, 49, 50, 54, 60, 64],
 'Setor C': [23, 26, 26, 29, 34, 35, 36, 41, 52, 56],
 'Setor D': [19, 20, 25, 27, 34, 39, 42, 44, 50, 65]}
Copia el código
Dado que cada sector tiene 10 colaboradores, construye un código que calcule la media de edad de cada sector, la edad media general entre todos los sectores y cuántas personas están por encima de la edad media general.

Si necesitas ayuda, las soluciones a las actividades están disponibles en la sección "Opinión del instructor".

Ver opinión del instructor
Opinión del instructor

Pueden existir diversas formas de resolver los problemas propuestos y algunas de ellas se presentan a continuación.

Calentando motores en la programación

1 - Incluso si ya sabemos manualmente cuál es el resultado, es interesante pensar cómo podríamos ejecutar las operaciones mediante programación.

# Datos de gastos
gastos = [2172.54, 3701.35, 3518.09, 3456.61, 3249.38, 2840.82, 3891.45, 3075.26, 2317.64, 3219.08]

# Calculamos el promedio encontrando el valor total de gastos con la función sum
# y la cantidad total de compras realizadas con len
total_gastos = sum(gastos)
cantidad_compras = len(gastos)
media_gastos = total_gastos / cantidad_compras
# Resultado
print(f'El promedio de gastos es {media_gastos} reales.')
Copia el código
2 -

# Datos de gastos
gastos = [2172.54, 3701.35, 3518.09, 3456.61, 3249.38, 2840.82, 3891.45, 3075.26, 2317.64, 3219.08]

# Variable que contará cuántas compras se realizaron por encima de 3000
contador_acima_3000 = 0
# Usamos un bucle para recorrer la lista de gastos
for gasto in gastos:
  # Verificamos si el elemento está por encima de 3000
  if gasto > 3000:
    # Sumamos uno al contador si hay algún valor por encima de 3000
    contador_acima_3000 += 1

# Con el conteo podemos calcular el porcentaje de valores por encima de 3000 entre todas las compras
porcentaje_acima_3000 = 100 * contador_acima_3000 / cantidad_compras

# Resultado
print(f'{contador_acima_3000} compras estuvieron por encima de R$3000,00.')
print(f'{porcentaje_acima_3000}% de los gastos estuvieron por encima de R$3000,00.')
Copia el código
3 -

# Lista que almacenará los 5 números enteros
lista_numeros = []

# Creamos un bucle que iterará 5 veces para recibir los 5 números
for i in range(0, 5):
  # Recopilamos el valor e lo insertamos en la lista 5 veces
  numero = int(input('Ingresa un número entero: '))
  lista_numeros.append(numero)
#Resultado
print(f'Lista de números ingresados: {lista_numeros}')
Copia el código
4 - Dado que el objetivo es imprimir la lista de números enteros en orden inverso al ingresado, podemos utilizar la técnica de partición, en la que la lista se recorre desde el último elemento (usando el índice -1) hasta el primero. De esta manera, la lista se imprime en orden inverso al original, de modo que el último elemento es el primero en mostrarse, seguido por el penúltimo elemento y así sucesivamente.

# Lista que almacenará los 5 números enteros
lista_numeros = []

# Creamos un bucle que iterará 5 veces para recibir los 5 números
for i in range(0, 5):
  # Recopilamos el valor e lo insertamos en la lista 5 veces
  numero = int(input('Ingresa un número entero: '))
  lista_numeros.append(numero)
# Usamos la técnica de partición para imprimir el resultado
print(f'Lista de números invertida: {lista_numeros[::-1]}')
Copia el código
5 - Para este problema, necesitamos verificar si todos los números, por debajo de lo especificado por el usuario, son o no divisibles por otros valores además de sí mismos. Para hacer esto, podemos anidar un bucle dentro de otro. Usamos un bucle para recorrer los números enteros de 2 al número ingresado. Para cada número, otro bucle se utiliza para verificar si es divisible por números más pequeños que él. Si el número no es divisible por ninguno de estos números, se considera un número primo. Esta estructura se puede observar a continuación:

# Recopilamos el número
numero = int(input('Ingresa un número entero: '))
# Lista para almacenar los números primos
lista_primos = []
# Bucle que recorre todos los números por debajo del número ingresado
for num in range(2, numero):
  # Primo es una bandera que nos permite saber si el valor analizado es primo o no.
  primo = True
  # Probamos si todos los números por debajo del especificado en el primer bucle pueden dar una división exacta.
  for prueba_divisibles in range(2, num):
    if num % prueba_divisibles == 0:
      # Si es divisible por algún número, entendemos que el número no es primo y terminamos el bucle interno con break.
      primo = False
      break
  # La condición se convierte en el resultado booleano de primo: False. Ignoramos la condición True y ejecutamos el bloque del if.
  if primo:
    lista_primos.append(num)
# Resultado
print(f'Lista de números primos: {lista_primos}')
Copia el código
6 - Recopilamos el día, mes y año de una fecha y realizamos comprobaciones mes a mes, confirmando si en febrero (sea o no un año bisiesto) el día coincide con lo que es posible. También se analizan los meses que terminan en 31 o 30 días para evitar la existencia de fechas inválidas en los datos.

# Recopilamos la fecha
dia = int(input('Ingrese el día: '))
mes = int(input('Ingrese el mes: '))
año = int(input('Ingrese el año: '))

# Análisis de febrero
if mes == 2:
  # Verificamos si es o no un año bisiesto
  if año % 4 == 0 and (año % 400 == 0 or año % 100 != 0):
    dias_febrero = 29
  else:
    dias_febrero = 28
  # Verificamos si el día ingresado coincide con el máximo de días de febrero
  if dia >= 1 and dia <= dias_febrero:
    print('Fecha válida')
  else:
    print('Fecha inválida')
# Verificamos meses que terminan en 31 días
elif mes in [1, 3, 5, 7, 8, 10, 12]:
  if dia >= 1 and dia <= 31:
    print('Fecha válida')
  else:
    print('Fecha inválida')
# Verificamos meses que terminan en 30 días
elif mes in [4, 6, 9, 11]:
  if dia >= 1 and dia <= 30:
    print('Fecha válida')
  else:
    print('Fecha inválida')
# Si el mes no está entre 1 y 12
else:
  print('Fecha inválida')
Copia el código
7 -

# Lista de crecimiento de bacterias
bacterias_colonia = [1.2, 2.1, 3.3, 5.0, 7.8, 11.3, 16.6, 25.1, 37.8, 56.9]
# Lista que almacenará las tasas de crecimiento
porcentaje_crecimiento = []
# Recorremos los índices de 1 a 9 para comparar los valores actuales con los anteriores
for i in range(1, len(bacterias_colonia)):
  # Realizamos el cálculo: 100 * (muestra_actual - muestra_anterior) / (muestra_anterior)
  porcentaje = 100 * (bacterias_colonia[i] - bacterias_colonia[i-1]) / (bacterias_colonia[i-1])
  # Agregamos el resultado a la lista porcentaje_crecimiento
  porcentaje_crecimiento.append(porcentaje)
# Resultado
print(f'Porcentajes de crecimiento:\n{porcentaje_crecimiento}')
Copia el código
8 -

# Lista que contendrá los valores de IDs
ids = []
# Variables contadoras de dulces y amargos
dulce = 0
amargo = 0

# Creamos un bucle que iterará 10 veces para recopilar los 10 IDs
for i in range(0,10):
  # Recopilamos el ID y lo agregamos a la lista
  ids.append(int(input(f'Ingrese el ID {i+1}°: '))

# Leemos todos los elementos de la lista de IDs y los asignamos a id
for id in ids:
  # Verificamos si los elementos son pares o impares para llevar el conteo
  if id % 2 == 0:
    dulce += 1
  else:
    amargo += 1

# Resultado
print(f'Cantidad de productos dulces: {dulce}')
print(f'Cantidad de productos amargos: {amargo}')
Copia el código
9 -

# Inicializamos los datos
respuestas = []  # Lista para almacenar las respuestas
# Lista de respuestas correctas
gabarito = ['D', 'A', 'C', 'B', 'A', 'D', 'C', 'C', 'A', 'B']
nota = 0  # Acumulará la nota total

# Recopilamos las respuestas del estudiante
for i in range(0, 10):
  respuestas.append(input(f'Introduzca la respuesta a la pregunta {i + 1}: ').upper())

# Verificamos si las respuestas coinciden y las sumamos a la nota
for i in range(0, 10):
  if respuestas[i] == gabarito[i]:
    nota += 1

# Mostrando la nota final
print(f'Nota final: {nota}')
Copia el código
10 -

# Recopilamos la lista de temperaturas mensuales
temperaturas_mensuales = []
for i in range(0, 12):
  temperaturas_mensuales.append(float(input(f'Ingrese la temperatura promedio del mes {i+1}: '))
# Creamos una lista auxiliar para los nombres de los meses
meses = ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio', 'Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre']
# Calculamos el promedio
media_anual = sum(temperaturas_mensuales) / len(temperaturas_mensuales)

# Resultado
print('Temperaturas por encima del promedio en: ')
for i in range(0, 12):
  # Verificamos todas las temperaturas en comparación con la media anual
  if temperaturas_mensuales[i] > media_anual:
    # Dado que los índices de los meses corresponden a las temperaturas,
    # podemos imprimirlos con el mismo índice
    print(meses[i])
Copia el código
11 -

# Diccionario de ventas
datos_ventas = {'Producto A': 300, 'Producto B': 80, 'Producto C': 60, 'Producto D': 200, 'Producto E': 250, 'Producto F': 30}

# Inicializamos las variables
total_ventas = 0  # Sumará todas las ventas
producto_mas_vendido = ''  # Almacenará el nombre del producto más vendido
unidades_producto_mas_vendido = 0  # Almacenará la cantidad más alta de ventas

# Recorremos las claves y elementos del diccionario
for producto in datos_ventas.keys():
    # Sumamos el total de ventas
    total_ventas += datos_ventas[producto]
    # Verificamos si el valor de ventas actual (datos_ventas[producto]) es mayor que el valor almacenado en unidades_producto_mas_vendido
    # Cada vez que datos_ventas[producto] supere el valor en unidades_producto_mas_vendido,
    # la variable unidades_producto_mas_vendido será igual a datos_ventas[producto], asignando un nuevo valor
    # De manera similar, producto_mas_vendido también se actualiza con el producto actual
    if datos_ventas[producto] > unidades_producto_mas_vendido:
        unidades_producto_mas_vendido = datos_ventas[producto]
        producto_mas_vendido = producto
# Resultados
print(f'Total de ventas es {total_ventas}')
print(f'{producto_mas_vendido} es el más vendido')
Copia el código
12 -

# Diccionario de votos por diseño
votos = {'Diseño 1': 1334, 'Diseño 2': 982, 'Diseño 3': 1751, 'Diseño 4': 210, 'Diseño 5': 1811}

# Inicializamos las variables
total_votos = 0  # Sumará todos los votos
ganador = ''  # Almacenará el nombre del diseño ganador
voto_ganador = 0  # Almacenará la cantidad ganadora de votos

# Recorremos las claves y elementos del diccionario
for diseño, voto_diseño in votos.items():
    # Sumamos el total de votos
    total_votos += voto_diseño
    # Verificamos si el voto del diseño actual (voto_diseño) es mayor que el valor almacenado en voto_ganador
    # Cada vez que voto_diseño supere el valor en voto_ganador,
    # la variable voto_ganador será igual a voto_diseño, asignando un nuevo valor
    # De manera similar, el ganador también se actualiza con el diseño actual
    if voto_diseño > voto_ganador:
        voto_ganador = voto_diseño
        ganador = diseño
# Calculamos el porcentaje del diseño ganador
porcentaje = 100 * (voto_ganador) / (total_votos)

# Resultado
print(f'{ganador} es el ganador: ')
print(f'Porcentaje de votos: {porcentaje}%')
Copia el código
13 -

# Lista de salarios
salarios = [1172, 1644, 2617, 5130, 5532, 6341, 6650, 7238, 7685, 7782, 7903]
# Inicializamos las variables
diccionario_abonos = {}  # Diccionario de abonos
total_abono = 0  # Sumará todos los gastos en abonos
abonos_minimo = 0  # Almacenará la cantidad de abonos mínimos
mayor_abono = 0  # Almacenará el mayor valor de abono

# Recorremos toda la lista de salarios
for salario in salarios:
    # Calculamos el valor teórico del abono
    abono = salario * 0.1
    # Si el abono es inferior a 200,
    # ajustamos el valor del abono al mínimo (200)
    if abono < 200:
        abono = 200
    # Agregamos un nuevo dato en el diccionario con la clave abono
    diccionario_abonos[salario] = abono

# Recorremos todos los valores del diccionario de abonos
for abono in diccionario_abonos.values():
    # Contamos los salarios mínimos
    if abono == 200:
        abonos_minimo += 1
    # Verificamos si el abono leído es mayor que el valor almacenado en mayor_abono
    # Cada vez que el abono supere el valor de mayor_abono,
    # la variable mayor_abono será igual al abono, asignando un nuevo valor
    if abono > mayor_abono:
        mayor_abono = abono
    # Sumamos los abonos
    total_abono += abono
# Resultados
print(f'Abonos: {diccionario_abonos}')
print(f'Total de gastos en abonos: {total_abono}')
print(f'Número de empleados que recibieron el abono mínimo: {abonos_minimo}')
print(f'Mayor valor de abono: {mayor_abono}')
Copia el código
14 -

# Especificamos los datos para un diccionario
datos = {'Área Norte': [2819, 7236],
         'Área Este': [1440, 9492],
         'Área Sur': [5969, 7496],
         'Área Oeste': [14446, 49688],
         'Área Centro': [22558, 45148]}
# Inicializamos las variables
suma_media = 0  # Sumará todos los promedios
mayor_diversidad = ''  # Almacenará el área con la mayor diversidad
mayor_suma = 0  # Almacenará la mayor suma de especies
# Recorremos las claves y elementos del diccionario
for área, especies in datos.items():
    # Sumamos el número de especies en cada área utilizando la función sum
    suma_especies = sum(especies)
    # Calculamos el promedio dividiendo la suma de las especies entre la cantidad de especies
    media = suma_especies / len(especies)
    # Imprimimos
    print(f'El {área} tiene un promedio de {media} especies')
    # Verificamos si la suma de especies es mayor que el valor almacenado de mayor_suma
    # Cada vez que suma_especies supere el valor de mayor_suma,
    # la variable mayor_suma será igual a suma_especies, asignando un nuevo valor
    # De manera similar, mayor_diversidad también se actualiza
    if suma_especies > mayor_suma:
        mayor_suma = suma_especies
        mayor_diversidad = área
    # Sumamos los promedios
    suma_media += media
# La media total será la suma_media dividida por la cantidad de áreas
media_total = suma_media / len(datos)
print(f'Media general de especies: {media_total}')
print(f'Área con la mayor diversidad biológica: {mayor_diversidad}')
Copia el código
15 -

# Especificamos los datos para un diccionario
datos = {'Sector A': [22, 26, 30, 30, 35, 38, 40, 56, 57, 65],
        'Sector B': [22, 24, 26, 33, 41, 49, 50, 54, 60, 64],
        'Sector C': [23, 26, 26, 29, 34, 35, 36, 41, 52, 56],
        'Sector D': [19, 20, 25, 27, 34, 39, 42, 44, 50, 65]}
# Inicializamos la variable que sumará todas las edades
total_edades = 0
# Recorremos las claves y elementos del diccionario
for sector, edades in datos.items():
  # Calculamos el promedio dividiendo la suma de las edades entre la cantidad de empleados en cada sector
  media_edad = sum(edades) / len(edades)
  # Imprimimos
  print(f'El {sector} tiene un promedio de {media_edad}')
  # Sumamos los promedios
  total_edades += sum(edades)
# La media total será el total_edades dividido por la cantidad de personas totales (sectores * empleados por sector)
media_total = total_edades / (len(edades) * len(datos))
print(f'La media de edad general es {media_total}')

# Inicializamos la variable que contará a todas las personas con edades por encima de la media
arriba_media = 0
# Recorremos nuevamente las claves y elementos del diccionario
for sector, edades in datos.items():
  # Leemos los elementos (edades) dentro de cada lista de edades en el diccionario
  for edad in edades:
    # Verificamos si el valor de la edad es superior a la media total
    if edad > media_total:
      # En caso de que el valor de la edad sea superior a la media, incrementamos en uno el contador
      arriba_media += 1
# Resultado
print(f'{arriba_media} personas están por encima de la edad media general')

Calentamiento

1 - Escribe un código para instalar la versión 3.7.1 de la biblioteca matplotlib.

2 - Escribe un código para importar la biblioteca numpy con el alias np.

3 - Crea un programa que lea la siguiente lista de números y elija uno al azar.

lista = [8, 12, 54, 23, 43, 1, 90, 87, 105, 77]
Copia el código
4 - Crea un programa que genere aleatoriamente un número entero menor que 100.

5 - Crea un programa que solicite a la persona usuaria ingresar dos números enteros y calcule la potencia del primer número elevado al segundo.

Aplicando a proyectos

6 - Se debe escribir un programa para sortear a un seguidor de una red social para ganar un premio. La lista de participantes está numerada y debemos elegir aleatoriamente un número según la cantidad de participantes. Pide a la persona usuaria que proporcione el número de participantes del sorteo y devuelve el número sorteado.

7 - Has recibido una solicitud para generar números de token para acceder a la aplicación de una empresa. El token debe ser par y variar de 1000 a 9998. Escribe un código que solicite el nombre de la persona usuaria y muestre un mensaje junto a este token generado aleatoriamente.

print(f"Hola, {nombre_usuario}, tu token de acceso es {token_generado} ¡Bienvenido/a!")
Copia el código
8 - Para diversificar y atraer nuevos clientes, una lanchonete creó un ítem misterioso en su menú llamado "ensalada de frutas sorpresa". En este ítem, se eligen aleatoriamente 3 frutas de una lista de 12 para componer la ensalada de frutas del cliente. Crea el código que realice esta selección aleatoria según la lista dada.

frutas = ["manzana", "banana", "uva", "pera", "mango", "coco", "sandia", "fresa", "naranja", "maracuya", "kiwi", "cereza"]
Copia el código
9 - Has recibido un desafío para calcular la raíz cuadrada de una lista de números, identificando cuáles resultan en un número entero. La lista es la siguiente:

numeros = [2, 8, 15, 23, 91, 112, 256]
Copia el código
10 - Haz un programa para una tienda que vende césped para jardines. Esta tienda trabaja con jardines circulares y el precio del metro cuadrado de césped es de R$ 25,00. Pide a la persona usuaria el radio del área circular y devuelve el valor en reales de cuánto tendrá que pagar.

A continuación presentamos las posibles respuestas al desafío propuesto:

1 - Escribe un código para instalar la versión 3.7.1 de la biblioteca matplotlib.

pip install matplotlib==3.7.1
Copia el código
2 - Escribe un código para importar la biblioteca numpy con el alias np.

import numpy as np
Copia el código
3 - Crea un programa que lea la siguiente lista de números y elija uno al azar.

import random

lista = [8, 12, 54, 23, 43, 1, 90, 87, 105, 77]
numero_aleatorio = random.choice(lista)
print(numero_aleatorio)
Copia el código
4 - Crea un programa que genere aleatoriamente un número entero menor que 100.

import random
numero_aleatorio = random.randrange(100)
print(numero_aleatorio)
Copia el código
5 - Crea un programa que solicite a la persona usuaria ingresar dos números enteros y calcule la potencia del primer número elevado al segundo.

import math
num1 = int(input("Ingrese el primer número entero: "))
num2 = int(input("Ingrese el segundo número entero: "))
resultado = math.pow(num1, num2)
print(f"La potencia de {num1} elevado a {num2} es {resultado}")
Copia el código
6 - Se debe escribir un programa para sortear a un seguidor de una red social para ganar un premio. La lista de participantes está numerada y debemos elegir aleatoriamente un número según la cantidad de participantes. Pide a la persona usuaria que proporcione el número de participantes del sorteo y devuelve el número sorteado.

import random

cantidad_participantes = int(input("Ingrese la cantidad de participantes: "))
numero_sorteado = random.randint(1, cantidad_participantes)
print(f"El número sorteado es: {numero_sorteado}")
Copia el código
7 - Has recibido una solicitud para generar números de token para acceder a la aplicación de una empresa. El token debe ser par y variar de 1000 a 9998. Escribe un código que solicite el nombre de la persona usuaria y muestre un mensaje junto a este token generado aleatoriamente.

import random

nombre_usuario = input("Ingrese su nombre: ")
token_generado = random.randrange(1000, 9999, 2)
print(f"Hola, {nombre_usuario}, tu token de acceso es {token_generado} ¡Bienvenido/a!")
Copia el código
8 - Para diversificar y atraer nuevos clientes, una lanchonete creó un ítem misterioso en su menú llamado "ensalada de frutas sorpresa". En este ítem, se eligen aleatoriamente 3 frutas de una lista de 12 para componer la ensalada de frutas del cliente. Crea el código que realice esta selección aleatoria según la lista dada.

import random

frutas = ["manzana", "banana", "uva", "pera", "mango", "coco", "sandia", "fresa", "naranja", "maracuya", "kiwi", "cereza"]
seleccion_frutas = random.sample(frutas, 3)
print("Ensalada de frutas sorpresa:", seleccion_frutas)
Copia el código
9 - Has recibido un desafío para calcular la raíz cuadrada de una lista de números, identificando cuáles resultan en un número entero. La lista es la siguiente:

import math

numeros = [2, 8, 15, 23, 91, 112, 256]
raices_enteras = [num for num in numeros if math.sqrt(num) % 1 == 0]
print("Números con raíces enteras:", raices_enteras)
Copia el código
10 - Haz un programa para una tienda que vende césped para jardines. Esta tienda trabaja con jardines circulares y el precio del metro cuadrado de césped es de R$ 25,00. Pide a la persona usuaria el radio del área circular y devuelve el valor en reales de cuánto tendrá que pagar.

import math

precio_metro_cuadrado = 25.00
radio = float(input("Ingrese el radio del área circular en metros: "))
area = math.pi * math.pow(radio, 2)
costo_total = precio_metro_cuadrado * area

print(f"El costo total es de R$ {round(costo_total,2)}")